src/main/java/org/folio/edge/sip2/repositories/CirculationRepository.java

```diff
@@ -109,71 +107,143 @@ public Future<CheckinResponse> performCheckinCommand(Checkin checkin, SessionDat
+          .map(resource -> {
+            final Optional<JsonObject> response = Optional.ofNullable(resource.getResource());
+
+            final OffsetDateTime dueDate = response
+              .map(v -> v.getString("dueDate"))
+              .map(v -> OffsetDateTime.from(Utils.getFolioDateTimeFormatter().parse(v)))
+              .orElse(OffsetDateTime.now(clock));
+
+            return CheckoutResponse.builder()
+              .ok(Boolean.valueOf(response.isPresent()))
+              .renewalOk(FALSE)
+              .magneticMedia(null)
+              .desensitize(Boolean.valueOf(response.isPresent()))
+              .transactionDate(OffsetDateTime.now(clock))
+              .institutionId(institutionId)
+              .patronIdentifier(patronIdentifier)
+              .itemIdentifier(itemIdentifier)
+              .titleIdentifier(response.map(
+                  v -> getChildString(v, "item", "title", UNKNOWN))
+                .orElse(resource.getTitle()))
+              .dueDate(dueDate)
+              .screenMessage(Optional.of(resource.getErrorMessages())
+                .filter(v -> !v.isEmpty())
+                .orElse(null))
+              .build();
+
+          });
+      });
+  }
+
+  private Future<IResource> addTitleIfNotFound(SessionData sessionData,
+                                               String itemIdentifier, IResource circRes) {
+    if (circRes.getErrorMessages().isEmpty()) {
+      return Future.succeededFuture(circRes);
+    }
+    return getTitle(itemIdentifier, sessionData, circRes.getErrorMessages());
+  }
+
+  private IResource getiResourceFromTitle(String title, List<String> circErrorMessages) {
+    IResource res;
+    res = new IResource() {
+      @Override
+      public JsonObject getResource() {
+        return null;
+      }
+
+      @Override
+      public String getTitle() {
+        return title;
+      }
+
+      @Override
+      public List<String> getErrorMessages() {
+        List<String> tempError = new ArrayList<>(circErrorMessages);
+        if(TITLE_NOT_FOUND.equals(title))
+        {
+          tempError.add("Title Not Found");
+          return tempError;
+        }
+        else
+          return tempError;
+      }
+    };
+    return res;
+  }
+
+  private Future<IResource> getTitle(String itemIdentifier, SessionData sessionData, List<String> circErrorMessages) {
+
+    final Map<String, String> headers = getBaseHeaders();
+    final ItemRequestData itemRequestData =
+      new ItemRequestData(null, headers, sessionData, itemIdentifier);
+
+    final Future<IResource> result = resourceProvider.retrieveResource(itemRequestData);
+
+    return result
+      .otherwise(Utils.handleSearchErrors(result.cause(), circErrorMessages))
+      .map(searchResult -> getTitelFromJson(searchResult, circErrorMessages));
+  }
+
+
+  private IResource getTitelFromJson(IResource resource, List<String> circErrorMessages) {
+    if (!resource.getErrorMessages().isEmpty())
```

Comment:
Spelling `Tite`

Reply:
Looks like the spelling error is still there. Should be `getTitleFromJson` not `getTiteFromJson`.